subgit/core.py

```diff
@@ -486,6 +487,111 @@ def pull(self, names):
+        """
+
+        config = self._get_config_file()
+
+        active_repos = self._get_active_repos(config)
+
+        if repo_names is None:  # If no names are specified in the command
+            active_repos_dict = config.get('repos', [])
+
+            repo_choices = ", ".join(active_repos_dict)
+
+            answer = self.yes_no(f"Are you sure you want to delete the following repos '{repo_choices}'")
+
+            if answer:
+                self.repo_deletion(active_repos=active_repos)
+
+        if isinstance(repo_names, list):  # If at least one name was specified
+
+            repo_choices = ", ".join(repo_names)
+
+            answer = self.yes_no(f"Are you sure you want to delete the following repos '{repo_choices}'")
+
+            if answer:
+                self.repo_deletion(active_repos, repo_names)
+
+    def repo_deletion(self, active_repos, repos=None):
+        """
+        Helper method that recieves a list of repos to delete and takes care of the logic
+        """
+        if not repos:
+            repos = active_repos
+
+        in_cnf_file = True
+
+        dirty_repos = False
+
+        repo_paths = []
+
+        path_no_exist = []
+
+        good_repos = []
+
+        for name in repos:
+            repo_paths.append(os.path.join(os.getcwd(), name))
+            if name not in active_repos:
+                log.warning(f"Argument: '{name}' does not exist in '.subgit.yml'")
+                in_cnf_file = False
+
+        if not in_cnf_file:
+            return 1
+
+        for path in repo_paths:
+
+            if not os.path.exists(path):
+                log.warning(f'Path to repo does not exist: {os.path.basename(path)} | Skipping {os.path.basename(path)}')
+                path_no_exist.append(path)
+                repo_paths.remove(path)
+
+        for path in repo_paths:
+
+            if not self.is_valid_repo(path):
+                log.warning(f"FATAL: '{os.path.basename(path)}' is not a git repo")
+                return 1
+
+        for path in repo_paths:
+
+            current_repo = Repo(path)
+
+            if self.check_remote(current_repo, os.path.basename(path)):
+                dirty_repos = True
+            else:
+                good_repos.append(path)
+
+        if not dirty_repos:
+            for repo in good_repos:
+                if path not in path_no_exist:
+                    shutil.rmtree(repo)
+                    log.info(f'Successfully removed repo: {os.path.basename(repo)}')
+
+    def check_remote(self, repo, base_name):
+        """
+        Takes repo object and name of directory to delete
+        """
+        is_diff = False
+        # has_new = repo.remotes.origin.refs["master"].commit != repo.heads[branch].commit
+        for remote in repo.remotes:
+            for branch in repo.branches:
+                if remote.refs[str(branch)].commit != repo.heads[str(branch)].commit or repo.is_dirty(untracked_files=True):
```

Comment:
A tip here when you get really complex but also very wide if cases is to create tmp variables ahead of it to reduce the if case a bit

```suggestion
                a = remote.refs[str(branch)].commit
                b = repo.heads[str(branch)].commit or repo.is_dirty(untracked_files=True)
                
                if a != b:
```

you would have to produce better variable names ofc. Also note that when you have multiple `and/or/not/!=` on the same if case, for readareadability for us users, you kinda must add in ( ) to explicitly show where you intend the logic to execute